### PR TITLE
5pm b - Schedule javascript api

### LINKED ISSUE #53: CRUD Operations on schedules (Front End)
User Story

As a student I can create and edit named schedules so that I can plan my classes.
Acceptance Criteria

- [x] I should be able to create new schedules and set their names. 
- [x] There is a display page for users to see their schedules.
- [x] There is a way to delete entire schedules

Implementation Todos

Front end:

- [x] There is a place to fill in the fields for a new schedule.
- [x] There is a create button for the schedule that creates a schedule in the backend.
- [x] There is a delete button for the schedule.
- [x] There is a field to get a schedule.

Testing:

- [x] The correct schedule has been created.
- [x] The correct schedule has been fetched.
- [x] The correct schedule has been deleted.

### PR DESCRIPTION
**Pull request by Mitchell57**
Adds an API for the front end to send requests to the database (create, read, and delete Schedules), as well as testing for the API

Partially closes #53

### CODE REVIEW

**Review by yiweizhangcs — APPROVED** [review]

**Review by andyyyho — APPROVED** [review]
Looks good

File: javascript/src/test/services/scheduleAPI.test.js [hunk 1]
```diff
@@ -0,0 +1,266 @@
...
+        name: "main",
+        description: "test",
+        userId: "123",
+        quarter: "W20"
+    };
+
+    const onSuccess = jest.fn();
+    const onError = jest.fn();
+
+    const result = fetchcreateScheduleJSON(expectedFields,useAuth0,onSuccess,onError);
```
Comment (btk5h) [inline javascript/src/test/services/scheduleAPI.test.js:41]: I don't think the value we pass to this function (and the other functions in your other test cases) is `useAuth0`, it's a function that returns the token. Can we make this less misleading by passing in a mock function instead?

File: javascript/src/main/services/scheduleAPI.js [hunk 2]
```diff
@@ -0,0 +1,81 @@
...
+
+//schedule
+const fetchcreateScheduleJSON = async (schedule, getToken, onSuccess, onError) => {
+    const token = await getToken();
+    const url = "/api/public/createSchedule?" + new URLSearchParams(schedule);
+    try {
+        const json = await fetchWithToken(url, getToken, {
+            method: "POST",
+            headers: {
+                Authorization: `Bearer ${token}`,
```
Comment (btk5h) [inline javascript/src/main/services/scheduleAPI.js:13]: Is this necessary (in this function and in the other functions in this file)? Doesn't `fetchWithToken` handle this for us?

File: javascript/src/test/services/scheduleAPI.test.js (hunk 1 above)
Comment (andyyyho) [reply javascript/src/test/services/scheduleAPI.test.js]: Thanks for the input, I just made a revision to account for this!

File: javascript/src/main/services/scheduleAPI.js (hunk 2 above)
Comment (andyyyho) [reply javascript/src/main/services/scheduleAPI.js]: I believe you are correct, I removed the lines and retested everything and it looks like it should all be in order now. Thanks for the input!

Comment (phtcon) [reply javascript/src/main/services/scheduleAPI.js]: I agree with @btk5h ... take a look at the implementation of `fetchWithToken` and make sure you are not reinventing the wheel.

File: javascript/src/main/services/scheduleAPI.js [hunk 3]
```diff
@@ -0,0 +1,75 @@
+import { fetchWithToken } from "main/utils/fetch"
+
+
+
+//schedule
+const fetchcreateScheduleJSON = async (schedule, getToken, onSuccess, onError) => {
```
Comment (phtcon) [inline javascript/src/main/services/scheduleAPI.js:6]: The prefix `fetch` and suffix `JSON` on each of these functions seems unnecessary and is out of step with the rest of the code base.  I wonder if we could just call each of these `createSchedule`, `deleteSchedule`, `getSchedule` and `getSchedules`?

Comment (phtcon) [reply javascript/src/main/services/scheduleAPI.js:6]: Otherwise, looks good to me!

**Review by phtcon — APPROVED** [review]
I have some quibbles with the naming, and would like to see it streamlined.  But I'm willing to let that go.

Other than that, LGTM

**Comment by phtcon** [comment]
20 points

**Review by andyyyho — DISMISSED** [review]
Looks good

File: javascript/src/main/services/scheduleAPI.js (hunk 3 above)
Comment (andrewhlu) [reply javascript/src/main/services/scheduleAPI.js:6]: @andyyyho I saw you put in an approval, but didn't resolve this conversation. Can you (or another team member) let us know whether you plan to make this change? And if not, could you fix the name to be correctly camelCased?

**Comment by andrewhlu** [comment]
Please address the comment above before we merge this PR

File: javascript/src/main/services/scheduleAPI.js (hunk 3 above)
Comment (andyyyho) [reply javascript/src/main/services/scheduleAPI.js]: @andrewhlu Sorry about that! I just fixed the camel casing and pushed it to this branch.

**Comment by pconrad** [comment]
Closing without merging; made obsolete by code in a later pull request.

**Pull request closed without merging**